apps/emqx_gateway/src/emqx_gateway_api_clients.erl

```diff
@@ -234,8 +238,17 @@ subscriptions(delete, #{
     }
 }) ->
     with_gateway(Name0, fun(GwName, _) ->
-        _ = emqx_gateway_http:client_unsubscribe(GwName, ClientId, Topic),
-        {204}
+        case lookup_topic(GwName, ClientId, Topic) of
+            {ok, _} ->
+                case emqx_gateway_http:client_unsubscribe(GwName, ClientId, Topic) of
+                    {error, not_found} ->
+                        return_http_error(404, "Client not found");
```

Comment:
Just a thought: maybe simplify this to `Subscription not found`, without trying to find out if it's client or topic that's not found?

Reply:
Wasn’t sure about this since it’s implementation dependent and there’s no strict contract. Some might return `{error, not_found}` if there’s no such subscription, others might not. Haven’t read through all the code tbh